### DISCUSSION TITLE
Where is the Documentation for grblHAL API?

### DISCUSSION THREAD

**Discussion by bear-fighter** [grblHAL/core#249] [Q&A] [answered]
I've scoured the wiki but haven't been able to find documentation for the API or information on what settings I'd need to input for my custom machine.  Any references to documentation would be appreciated.

**Comment by terjeio** [comment, marked as answer]
The driver API documentation can be found [here](http://svn.io-engineering.com/grblHAL/html/hal_8h.html), and defaults for core settings [here](http://svn.io-engineering.com/grblHAL/html/config_8h.html). Note that the API documentation is very much work in progress - some missing documentation can be found in the source files but not all.

Run time settings documentation can be found partially in the [grblHAL Wiki](https://github.com/grblHAL/core/wiki/Additional-or-extended-settings), and partially in the legacy [Grbl Wiki](https://github.com/gnea/grbl/wiki/Grbl-v1.1-Configuration). Settings related to plugins are mostly documented in the plugin readme, and all has information available from the controller itself via $-commands. Use `$help` from a terminal for info.

**Reply by bear-fighter** [reply]
Perfect, thank you for the pointers.  As I work through setting up my machine (using rp2040 driver), I'm planning on taking notes/documenting the setup.  Is there a good way to submit or merge those notes with existing documentation?
